app/src/main/java/com/android/sample/model/profile/ProfileViewModel.kt

```diff
@@ -32,10 +37,32 @@ open class ProfileViewModel @Inject constructor(private val repository: Profiles
   }
 
   fun fetchUserData(userId: String) {
```

Comment:
Modifying the fetchUserData function to cache user data locally on successful Firestore fetch and load from Room if Firestore fails.